airbyte-cdk/python/airbyte_cdk/sources/declarative/extractors/record_filter.py

```diff
@@ -64,11 +64,7 @@ def _start_date_from_config(self) -> datetime.datetime:
 
     @property
     def _end_datetime(self) -> datetime.datetime:
-        return (
-            self._date_time_based_cursor._end_datetime.get_datetime(self._date_time_based_cursor.config)
-            if self._date_time_based_cursor._end_datetime
-            else datetime.datetime.max
-        )
+        return self._date_time_based_cursor._select_best_end_datetime()
```

Comment:
If we're going to start using this private method `_select_best_end_datetime()` we should expose this publicly on the `DateTimeBasedCursor`

Reply:
updated, it is public now